rust/src/jsonbuilder.rs

```diff
@@ -103,8 +104,9 @@ pub struct JsonBuilderMark {
     state: u64,
 }
 
+// JsonBuilder should not be made public outside the crate as it is not repr(C)
 #[derive(Debug, Clone)]
-pub struct JsonBuilder {
+pub(crate) struct JsonBuilder {
```

Comment:
Might not be the only case case for Suricata as a library.  Could have a Rust crate pulling Suricata for utility reasons and need access.

Reply:
Indeed, but how do you propose to avoid the pitfall that some plugin use suricata crate JsonBuilder instead of using the suricata runtime one ? Just rely on docs ?

Reply:
Probably through documentation.  I'm not too keen on artificially limiting the usefulness of the library to protect users of just one use case.